README.md

```diff
@@ -2217,6 +2218,151 @@ RMB will return a response to the client as follows:
 RMB will not cross check the raml to see that these statuses have been defined for the endpoint. This is the developer's responsibility.
 
 
+## JSON Schema migration for module update
+
+The module update adds a few requirements to the database migration.
+We no longer need to just adapt the database in the way it’s required by the new version of the module; we also need to do it in a way that the old and the new version of the module can work with the database.
+That means that all migrations (schema changes) need to be backward-compatible to allow to perform a seamless rollback (database backup-restore could be quite expensive operations to perform it frequently) to the previous module’s version in case something goes wrong during the update operation.
+But not all operations, e.g., renaming or removing a property from JSON schema or adding a property with a default value, are backward compatible.
+These operations require a multi-step process that enables us to perform the migration without breaking our system.
+
+A general approach for roll out a new version for RMB based modules (platform-core, platform-complete)
+* Register a module descriptor for the new version of the module using POST /_/proxy/modules
+* Deploy a new version of the module using POST /_/discovery/modules
+* Install the new version of the module for a tenant using POST /_/proxy/tenants/{tenant}/install
+* Check the log of the module for any errors or issues.
+
+For RMB based modules a DB migration will be done automatically based on the content of the schema.json provided within the module. Indeed there are exceptions, but each such case must be considered separately.
+
+
+#### Recommendations for the creation and maintenance of the schema.json file
+* For each table in the “tables” section provide “fromModuleVersion” value even if this table has not been added in this version and already exists in the database. It will eliminate the generation and running of unnecessary SQL statements for the table. You can provide a value for the first production version for example
+    ```
+      {
+        "tableName": "loan_type",
+        "fromModuleVersion": "mod-inventory-storage-1.0.0",
```

Comment:
Does `mod-inventory-storage-1.0.0-SNAPSHOT.265` match this from version?

Reply:
fromModuleVersion - this field indicates the version in which the table was created / updated in. The same for scripts. ComparableVersion.java forked from maven is used under the hood for comparison. So  mod-inventory-storage-1.0.0-SNAPSHOT.265 is a valid version number, 

This is a result of a comparison between  mod-inventory-storage-1.0.0-SNAPSHOT.265 and mod-inventory-storage-1.0.0-SNAPSHOT
Display parameters as parsed by Maven (in canonical form) and comparison result:
1. mod-inventory-storage-1.0.0-SNAPSHOT.265 == mod-inventory-storage-1-snapshot.265
   mod-inventory-storage-1.0.0-SNAPSHOT.265 > mod-inventory-storage-1.0.0-SNAPSHOT
2. mod-inventory-storage-1.0.0-SNAPSHOT == mod-inventory-storage-1-snapshot

Reply:
I don't think I follow. Maybe an example could help me.

If we have a schema entry that states a `from version` of `mod-inventory-storage-1.0.0`

If this was introduced in the build `mod-inventory-storage-SNAPSHOT.274`

What should happen during the tenant API when the `toVersion` is `mod-inventory-storage-SNAPSHOT.274` on a fresh system?

Will the schema migration that states a `from version` of `mod-inventory-storage-1.0.0` be executed?

Reply:
Let’s look at a case
-	The current version of a module enabled for a tenant is mod-inventory-storage-SNAPSHOT.274
-	The version of a module that will be enabled for a tenant (it is provided in the ModuleDescriptor.json) is mod-inventory-storage-SNAPSHOT.275
-	The version specified in the “fromModuleVersion” in schema.json is mod-inventory-storage-1.0.0
The comparison will be done between mod-inventory-storage-1.0.0 and mod-inventory-storage-SNAPSHOT.274 and the result is mod-inventory-storage-1.0.0 > mod-inventory-storage-SNAPSHOT.274 so a script/table will be applied/created or changed.

Another case 
-	The current version of a module enabled for a tenant is mod-inventory-storage-1.0.0-SNAPSHOT.274
-	The version of a module that will be enabled for a tenant (it is provided in the ModuleDescriptor.json) is mod-inventory-storage-1.0.0-SNAPSHOT.275
-	The version specified in the “fromModuleVersion” in schema.json is mod-inventory-storage-1.0.0
The comparison will be done between mod-inventory-storage-1.0.0 and mod-inventory-storage-1.0.0-SNAPSHOT.274 and the result is mod-inventory-storage-1.0.0 > mod-inventory-storage-1.0.0-SNAPSHOT.274 so a script/table will be applied/created or changed as well.

In both cases we have the correct behavior.